packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -870,7 +870,12 @@ export function cloneElement(element, config, children) {
         // Silently steal the ref from the parent.
         ref = config.ref;
         if (!disableStringRefs) {
-          ref = coerceStringRef(ref, owner, element.type);
+          let refOwner = owner;
+          if (refOwner == null) {
+            // fall back to the current owner
+            refOwner = ReactSharedInternals.owner;
+          }
+          ref = coerceStringRef(ref, refOwner, element.type);
```

Comment:
I think the proper fix it to move line 881 above the `if (!disableStringRefs)` block? 

This comment suggests that the current owner should be used whenever a new ref is provided:

```js
  // Owner will be preserved, unless ref is overridden
```

Reply:
```diff
    if (hasValidRef(config)) {
+     owner = ReactSharedInternals.owner;
      if (!enableRefAsProp) {
        // Silently steal the ref from the parent.
        ref = config.ref;
        if (!disableStringRefs) {
          ref = coerceStringRef(ref, owner, element.type);
        }
      }
-     owner = ReactSharedInternals.owner;
    }
```

Reply:
yeah that seemed like the ideal fix but i figured i'd start conservative until checking with you